### ISSUE TITLE
"ignore-this" option for "prefer-destructuring"

### ISSUE THREAD

**Issue by wparrott-touchnet** [eslint/eslint#10996]
**What rule do you want to change?**
`prefer-destructuring`

**Does this change cause the rule to produce more or fewer warnings?**
fewer

**How will the change be implemented? (New option, new default behavior, etc.)?**
new option

**Please provide some example code that this change will affect:**

```js
class MyComponent extends React.Component {
    // ...
    render() {
        const {prop1, prop2} = this.props;
        // ...
    }
}
```

**What does the rule currently do for this code?**
Right now, this rule would report the above code as a warning or error. The fix would require the code to look as follows:

```js
class MyComponent extends React.Component {
    // ...
    render() {
        const {props: {prop1, prop2}} = this;
        // ...
    }
}
```

**What will the rule do after it's changed?**
I'd like to see an option (default could be the current behavior for compatibility) to allow the behavior in the first example. To me, it's clearer than the second snippet. I'd like to enforce destructuring as a general rule/recommendation across our teams' projects, but the existing rule will cause huge amounts of warnings which, to me, are essentially false positives.

**Comment by kaicataldo** [comment]
I think the `prefer-destructuring` rule is a bit kludgy in its current iteration, and I think some option like this would make it more usable.

That being said, I feel like scoping the change to properties on `this` is a bit of a narrow use case. Thoughts on adding an option that allows this allows this style for any member expression? Something like `onlyEnforceForLastMember` (hopefully there's a better name than that)? 

```js
/* eslint prefer-destructuring: ["error", { onlyEnforceForLastMember: true }] */

// Valid
const { foo } = this.bar;
const { bar: { foo } } = this;
const { foo } = bar.baz;
const { bar: { foo } } = baz;

const [foo] = this.bar;
const { bar: [foo] } = this;
const [foo] = bar.baz;
const { bar: [foo] } = baz;

// Invalid 
const foo = this.bar.foo;
const foo = bar.baz.foo;
const foo = this.bar[0];
const foo = bar.baz[0];
```

**Comment by ljharb** [comment]
For airbnb’s use case, we would only want this option for `this` - the receiver is special, and warrants special treatment.

**Comment by wparrott-touchnet** [comment]
Seeing the discussion here, I can actually see a use case for both variations of the flag. Perhaps something along these lines would satisfy both situations?

```
"prefer-destructuring": [
  "error",
  // (snip)
  {
    "enforceForRenamedProperties": true|false,
    "enforceForLastMemberOnly": true|false,
    "ignoreThisTerm": true|false
  }
]
```
